GitHubData/Services/IRequestRateLimitService.cs

```diff
@@ -0,0 +1,24 @@
+// <copyright file="IRequestRateLimitService.cs" company="KinsonDigital">
+// Copyright (c) KinsonDigital. All rights reserved.
+// </copyright>
+
+namespace GitHubData.Services;
+
+/// <summary>
+/// Provides the ability to get request rate limit data for
+/// non-authorized and authorized requests to the GitHub API.
+/// </summary>
+public interface IRequestRateLimitService
+{
+    /// <summary>
+    /// Gets or sets the auth token to use when making requests for GitHub data.
+    /// </summary>
+    /// <remarks>If the token is null or empty, then non-authorized requests will be made.</remarks>
```

Comment:
Refactor code docs to below:

```suggestion
    /// <remarks>If the token is null or empty, then non-authorized requests will be used.</remarks>
```